core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -576,8 +576,8 @@ private[zio] final class FiberContext[E, A](
                   case ZIO.Tags.FiberRefNew =>
                     val zio = curZio.asInstanceOf[ZIO.FiberRefNew[Any]]
 
-                    val fiberRef = new FiberRef[Any](zio.initialValue, zio.combine)
-                    fiberRefLocals.put(fiberRef, zio.initialValue)
+                    val fiberRef = new FiberRef[Any](zio.initial, zio.onFork, zio.onJoin)
+                    fiberRefLocals.put(fiberRef, zio.onFork(zio.initial))
```

Comment:
Wouldn't the right place to put this be where a child fiber accesses a parent fiber's ref for the first time?

This is calling 'fork' whenever a new fiber ref is created. But we would like it just to be called during "handoff" of a fiber ref from parent to child.

Reply:
Yes. Fixed now.